cookiecutter-app/{{cookiecutter.directory_name}}/app.py

```diff
@@ -1,37 +1,54 @@
 # -*- coding: utf-8 -*-
 
-from __future__ import absolute_import
+from typing import cast
+from flask import current_app
 
 from eduid_common.api.app import eduid_init_app
+from eduid_common.api import mail_relay
 from eduid_common.api import am, msg
+from eduid_common.authn.middleware import AuthnApp
+from eduid_webapp.{{cookiecutter.directory_name}}.settings.common import {{cookiecutter.class_name}}Config
 
 __author__ = '{{cookiecutter.author}}'
 
 
-def init_{{cookiecutter.directory_name}}_app(name, config=None):
+class {{cookiecutter.class_name}}App(AuthnApp):
+
+    def __init__(self, name, config):
+        # Init config for common setup
+        config = get_app_config(name, config)
+        super({{cookiecutter.class_name}}App, self).__init__(name, config)
+        # Init app config
+        self.config = {{cookiecutter.class_name}}Config(**config)
+        # Init dbs
+        self.private_userdb = {{cookiecutter.class_name}}UserDB(self.config.mongo_uri)
+        # Init celery
+        msg.init_relay(self)
+        am.init_relay(self, 'eduid_{{cookiecutter.directory_name}}')
+        # Initiate external modules
+
+
+def get_current_app() -> {{cookiecutter.class_name}}App:
+    """Teach pycharm about {{cookiecutter.class_name}}App"""
+    return current_app  # type: ignore
+
+
+current_{{cookiecutter.directory_name}}_app = get_current_app()
```

Comment:
I'm not sure I understand this. Will this not just remove `current_app` (and therefore `current_app.config`) from mypy's checks - effectively making configuration untyped?

Reply:
To answer myself: nope =)